confluent_kafka/avro/__init__.py

```diff
@@ -98,9 +98,11 @@ class AvroConsumer(Consumer):
     Constructor takes below parameters
 
     :param dict config: Config parameters containing url for schema registry (``schema.registry.url``)
-                        and the standard Kafka client configuration (``bootstrap.servers`` et.al).
+                        and the standard Kafka client configuration (``bootstrap.servers`` et.al)
+    :param optional a reader schema for the message key
+    :param optional a reader schema for the message value
     """
-    def __init__(self, config, schema_registry=None):
+    def __init__(self, config, schema_registry=None, reader_key_schema=None, reader_value_schema=None):
```

Comment:
I had it first that way (or `key_schema_reader` considering `reader` as var index), but this would be inconsistent with `default_key_schema` and `default_value_schema` in the constructor: 

https://github.com/confluentinc/confluent-kafka-python/blob/3b69697a7500e3dcf888c801dfa7d4edf8ae6c0a/confluent_kafka/avro/__init__.py#L29

Furthermore I consider `key_schema`/`value_schema` compounds naming an attribute/member of the object/namespace `reader`

This is also more consistent with `fastavro` (`writer_schema`, `reader_schema`)
https://fastavro.readthedocs.io/en/latest/reader.html#fastavro._read_py.schemaless_reader

and the REST Proxy (`key_schema` and `value_schema`)
https://docs.confluent.io/current/kafka-rest/docs/api.html#post--topics-(string-topic_name)

Reply:
I had it first that way (or key_schema_reader considering reader as var index), but this would be inconsistent with default_key_schema and default_value_schema in the constructor:

See https://github.com/confluentinc/confluent-kafka-python/blob/443177e1c83d9b66ce30f5eb8775e062453a738b/confluent_kafka/avro/__init__.py#L29-L30

Furthermore I consider `key_schema`/`value_schema` compounds naming an attribute/member of the object/namespace reader

This is also more consistent with fastavro (`writer_schema`, `reader_schema`)
https://fastavro.readthedocs.io/en/latest/reader.html#fastavro._read_py.schemaless_reader

and the REST Proxy (`key_schema` and `value_schema`)
https://docs.confluent.io/current/kafka-rest/docs/api.html#post--topics-(string-topic_name)